### PR TITLE
[WIP] Fix bug in internal PBF deserialization of locations

### LINKED ISSUE #1448: Location type information is lost when deserializing from PBF
When switching from JSON to PBF for internal communication, location type information was not correctly deserialized. Furthermore, the nature of the bug ensured that they always ended up as `through`. The change is imperceptible in most cases (since code exists elsewhere to unconditionally make the first and last locations `break`s), but it makes it impossible to request a route with multiple legs.

### LINKED ISSUE #1468: Break locations treated as through
Hi. I'm trying to get a route with multiple breaks, however it seems it always treats the intermediate locations as "through" and always returns a single leg.

Version: 2.6.3

Request:
```json
{
  "locations":[
    {"lat":41.285488,"lon":-8.650597,"type":"break"},
    {"lat":41.434488,"lon":-8.230597,"type":"break"},
    {"lat":41.994025,"lon":-7.860695,"type":"break"}
  ],
  "costing":"auto"
}
```

Response:
```json
...
        "locations": [
            {
                "original_index": 0,
                "type": "break",
                "lon": -8.650597,
                "lat": 41.285488,
                "side_of_street": "right"
            },
            {
                "original_index": 1,
                "lon": -8.230597,
                "lat": 41.434486,
                "type": "through"
            },
            {
                "original_index": 2,
                "lon": -7.860695,
                "lat": 41.994026,
                "type": "break"
            }
        ],
...
```

### PR DESCRIPTION
**Pull request by ianthetechie**
# Issue

This will close #1448 and it will close #1468 

I'll try to write up a test for this in the next day or two, and have marked it WIP until then. I didn't see a CONTRIBUTING doc (just a paragraph or 2 in the main README), so let me know if I am missing anything.

## Tasklist

 - [x] Add tests
 - [x] Review - you must request approval to merge any PR to master
 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [ ] Update the [changelog](CHANGELOG.md)
 - [x] Update relevant [documentation](docs/README.md)

### CODE REVIEW

**Comment by ianthetechie** [comment]
Note that I was unable to get the tidy script to run on mac. I'll run it one way or another before removing the WIP flag.

**Comment by ianthetechie** [comment]
Hey @kevinkreiser mind taking a look now?

* I was able to get the tidy script to run on a Linux box.
* I wrote a test for the patch, and in the process touched one of the operator implementations and added a method since the original behavior seemed rather strange (fortunately, it appears to have only been used in a test, which incidentally documented the behavior better than the method docstring). If you'd prefer this be implemented differently, let me know. I identified usages by adding `delete =` to the operator to make sure this wouldn't have any unintended performance side-effects.
* My test does pass, but it looks like either `master` is busted, something externally changed, etc... I'm assuming this isn't my fault based on the string of red X's on the PR list, but if you need me to pull in some new changes or anything, let me know.
* I'd recommend backporting this to 2.x, as this issue affects all release builds since February.

**Comment by kevinkreiser** [comment]
related to #1468 i think

**Comment by kevinkreiser** [comment]
@ianthetechie i believe ive fixed the issue with the failing node tests :crossed_fingers: that CI passes as well

File: valhalla/baldr/pathlocation.h [hunk 1]
```diff
@@ -167,7 +173,8 @@ struct PathLocation : public Location {
 
   static Location fromPBF(const odin::Location& loc) {
     Location l({loc.ll().lng(), loc.ll().lat()},
-               odin::Location::kBreak ? Location::StopType::BREAK : Location::StopType::THROUGH,
```
Comment (kevinkreiser) [inline valhalla/baldr/pathlocation.h]: just for posterity, this is where the real bug lies. kBreak == 0 and therefore everything gets set to through. im pretty sure that this bug is my fault.. sorry all!

**Comment by blackgate** [comment]
I already checked this and it fixes the #1468
Thank you

**Comment by darbultr** [comment]
problem also exists on docker

**Comment by ianthetechie** [comment]
From the looks of it, dockerized builds are still very much WIP. I would recommend building the v2.7.0 from source.

**Comment by kevinkreiser** [comment]
just a freindly reminder, master is no longer unstable. feel free to build master from source instead of sticking on 2.7.0. and yeah we haven't given docker much love in a while.. need to find time to get back to that at some point!

**Pull request merged**